packages/std/src/math/decimal.rs

```diff
@@ -1195,6 +1211,119 @@ mod tests {
+
+        // Move right
+        let a = dec("123127726548762582");
+        assert_eq!(a / dec("1"), dec("123127726548762582"));
+        assert_eq!(a / dec("10"), dec("12312772654876258.2"));
+        assert_eq!(a / dec("100"), dec("1231277265487625.82"));
+        assert_eq!(a / dec("1000"), dec("123127726548762.582"));
+        assert_eq!(a / dec("1000000"), dec("123127726548.762582"));
+        assert_eq!(a / dec("1000000000"), dec("123127726.548762582"));
+        assert_eq!(a / dec("1000000000000"), dec("123127.726548762582"));
+        assert_eq!(a / dec("1000000000000000"), dec("123.127726548762582"));
+        assert_eq!(a / dec("1000000000000000000"), dec("0.123127726548762582"));
+        assert_eq!(dec("1") / a, dec("0.000000000000000008"));
+        assert_eq!(dec("10") / a, dec("0.000000000000000081"));
+        assert_eq!(dec("100") / a, dec("0.000000000000000812"));
+        assert_eq!(dec("1000") / a, dec("0.000000000000008121"));
+        assert_eq!(dec("1000000") / a, dec("0.000000000008121647"));
+        assert_eq!(dec("1000000000") / a, dec("0.000000008121647560"));
+        assert_eq!(dec("1000000000000") / a, dec("0.000008121647560868"));
+        assert_eq!(dec("1000000000000000") / a, dec("0.008121647560868164"));
+        assert_eq!(dec("1000000000000000000") / a, dec("8.121647560868164773"));
+
+        // Move left
+        let a = dec("0.123127726548762582");
+        assert_eq!(a / dec("1.0"), dec("0.123127726548762582"));
+        assert_eq!(a / dec("0.1"), dec("1.23127726548762582"));
+        assert_eq!(a / dec("0.01"), dec("12.3127726548762582"));
+        assert_eq!(a / dec("0.001"), dec("123.127726548762582"));
+        assert_eq!(a / dec("0.000001"), dec("123127.726548762582"));
+        assert_eq!(a / dec("0.000000001"), dec("123127726.548762582"));
+        assert_eq!(a / dec("0.000000000001"), dec("123127726548.762582"));
+        assert_eq!(a / dec("0.000000000000001"), dec("123127726548762.582"));
+        assert_eq!(a / dec("0.000000000000000001"), dec("123127726548762582"));
+
+        assert_eq!(
+            Decimal::percent(15) / Decimal::percent(60),
+            Decimal::percent(25)
+        );
+
+        // works for refs
+        let a = Decimal::percent(100);
+        let b = Decimal::percent(20);
+        let expected = Decimal::percent(500);
+        assert_eq!(a / b, expected);
+        assert_eq!(&a / b, expected);
+        assert_eq!(a / &b, expected);
+        assert_eq!(&a / &b, expected);
+    }
+
+    #[test]
+    fn decimal_div_assign_works() {
+        let mut a = Decimal::percent(15);
+        a /= Decimal::percent(20);
+        assert_eq!(a, Decimal::percent(75));
+
+        // works for refs
+        let mut a = Decimal::percent(50);
+        let b = Decimal::percent(20);
+        a /= &b;
+        assert_eq!(a, Decimal::percent(250));
+    }
+
+    #[test]
+    #[should_panic(expected = "Multiplication overflow")]
```

Comment:
I think we should have a better panic message for this case (by creating `checked_from_ratio` and having explicit panics in `fn div`)